### PR TITLE
Use sender in the router queries (2)

### PR DESCRIPTION
**Pull request by jubeira**
# Description

We're saving the sender but apparently in some cases it's important to actually pass the sender to the Vault.

## Type of change

- [x] Bug fix 
- [ ] New feature 
- [ ] Breaking change 
- [ ] Dependency changes
- [ ] Code refactor / cleanup
- [ ] Optimization: [ ] gas / [ ] bytecode
- [ ] Documentation or wording changes
- [ ] Other

## Checklist:

- [x] The diff is legible and has no extraneous changes
- [x] Complex code has been commented, including external interfaces
- [x] Tests have 100% code coverage
- [x] The base branch is either `main`, or there's a description of how to merge

## Issue Resolution

N/A

### CODE REVIEW

File: pkg/vault/contracts/Router.sol [hunk 1]
```diff
@@ -777,9 +777,7 @@ contract Router is IRouter, RouterCommon, ReentrancyGuardTransient {
                 abi.encodeCall(
                     Router.queryAddLiquidityHook,
                     AddLiquidityHookParams({
-                        // We use the Router as a sender to simplify basic query functions,
-                        // but it is possible to add liquidity to any recipient.
-                        sender: address(this),
+                        sender: sender,
```
Comment (joaobrunoah) [inline pkg/vault/contracts/Router.sol:780]: It'd be nice to have a mutation test, checking if each query hook is called with the right sender, but maybe for another PR.

**Review by joaobrunoah — APPROVED** [review]
LGTM, although there is a lint error in a hardhat test file.

File: pkg/vault/contracts/CompositeLiquidityRouter.sol [hunk 2]
```diff
@@ -1,6 +1,6 @@
 // SPDX-License-Identifier: GPL-3.0-or-later
 
-pragma solidity ^0.8.26;
+pragma solidity ^0.8.24;
```
Comment (EndymionJkb) [inline pkg/vault/contracts/CompositeLiquidityRouter.sol:3]: Was waiting for that to get fixed :)

**Review by EndymionJkb — APPROVED** [review]
Pending CI, LGTM. I don't see any missing ones.

**Comment by jubeira** [comment]
This is not the right approach; the integration problem is specific to the `CompositeLiquidityRouter`.

Superseded by #1072

**Pull request closed without merging**